Archlinux: AUR, ARM, distcc, multi stacks, ...

mickybart opened the pull request:
Support AUR packages
Cleanup output
Debug mode during the rootfs build
Add libhybris-git (installed by adbd on Ubuntu but not yet available for Arch so this is a first step)
rsync available by default (useful for some AUR packages)
Support build from ARM 32bit host
Support Distcc (useful to compile AUR packages)
Support comments into package-lists/*.chroot + independent installation per file
Support for different stacks (halium is used by default but some plasma-mobile, experimental or whatever stacks with dedicated hooks and package-lists can be set)

mickybart committed:
clean output, debug mode and AUR support

make DEBUG=1 supported
Remove every commands output by default
Rework hooks
Support AUR packages with an AUR helper

mickybart committed:
halium: add libhybris

mickybart committed:
add rsync as a base package

mickybart committed:
Makefile: support ARM host

mickybart committed:
distcc support

z3ntu commented on customization/hooks/aur-helper.sh:
```diff
@@ -0,0 +1,81 @@
+		pacaur)
+			git clone https://aur.archlinux.org/pacaur.git
+			cd pacaur
+			makepkg -s --install --noconfirm
```
pacaur depends on cower which is in the AUR too...

z3ntu replied:
Oh wait, I think ALARM has cower in the repos...

mickybart replied:
I confirm that this is on a repo for now so I'm relying on it :)

mickybart committed:
chroot: package-lists: Isolate each file

Installation based on the content of every files independently.
This permit to pass some flags to the package manager or to install some
packages before some others and to don't delay some post processing operations

Support of comments on package-lists/*.chroot files

mickybart committed:
Support for different stacks

Move aur-helper.sh and chroot-hooks.sh that are common to every stacks.
Those scripts are not real hooks and a builder folder is more appropriate for them.

Move current hooks and package-lists to "halium" stack

default stack is set to halium
stack can be overload with make STACK="<name>"

mickybart committed:
fix sort issue since the used of find

mickybart committed:
MAKEFLAGS configuration exposed to Makefile

vanyasem commented:
`pacaur` isn't maintained anymore, I guess it would be nice to replace it with some other AUR helper

mickybart commented:
Do you have an alternative in mind ? this is a sensitive topic ^^

vanyasem commented:
Just pick whatever has the most votes on AUR. `trizen` looks like a nice alternative. 

`yaourt` was known to be vulnerable, and `pacaur` is unmaintained (and recently got partially broken by a `pacman` update)
